CMakeLists.txt

```diff
@@ -123,6 +123,10 @@ else ()
     message (AUTHOR_WARNING "${ISSUE_REPORT_MSG}")
 endif ()
 
+if (NN_STATIC_LIB)
+    add_definitions (-DNN_STATIC_LIB)
+endif ()
+
```

Comment:
It's not particularly obvious why this fixes the issue; the reason is because the definition is applied prior to adding tests/tools. The reason it didn't work properly before is because this defintion only applied to the `src` directory -- meaning, effective only the library itself, but none of the companion executables within this project.